### ISSUE TITLE
Tensors as output of keras models instead of numpy arrays.

### ISSUE THREAD

**Issue by pratikgujjar** [keras-team/keras#12286]
Please make sure that the boxes below are checked before you submit your issue.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.

Thank you!

- [x] Check that you are up-to-date with the master branch of Keras. You can update with:
`pip install git+git://github.com/keras-team/keras.git --upgrade --no-deps`

- [x] Check that your version of TensorFlow is up-to-date. The installation instructions can be found [here](https://www.tensorflow.org/get_started/os_setup).

- [ ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

I have two keras models model_a and model_b. I need the output of model_a to calculate a custom loss against the output of model_b. However model_a.predict(X_train) returns a numpy array that I have to rewrite onto a keras.backend.variable to calculate the loss:

def loss(y_true, y_pred, model_a_output):
   return mse(y_true_y_pred) + kld(y_pred, model_a_output)

Is there a way to directly obtain model_a predictions as tensors so that all operations are retained in the GPU?
